dataland-frontend/src/components/forms/parts/elements/derived/NaceCodeSelector.vue

```diff
@@ -0,0 +1,210 @@
+              <span class="p-badge p-badge-no-gutter">{{ selectedChildrenCounter.get(slotProps.node.key) || 0 }}</span>
+            </div>
+            <span>{{ slotProps.node.label }}</span>
+          </div>
+        </template>
+      </Tree>
+    </div>
+  </OverlayPanel>
+</template>
+
+<script lang="ts">
+import Tree, { TreeNode } from "primevue/tree";
+import InputText from "primevue/inputtext";
+import OverlayPanel from "primevue/overlaypanel";
+import Checkbox from "primevue/checkbox";
+
+import { naceCodeTree, naceCodeMap, filterNodes } from "@/components/forms/parts/elements/derived/NaceCodeTree";
+import Tooltip from "primevue/tooltip";
+import { defineComponent, PropType, ref } from "vue";
+import { assertDefined } from "@/utils/TypeScriptUtils";
+
+export default defineComponent({
+  name: "NaceCodeSelector",
+  emits: ["update:modelValue"],
+  directives: {
+    tooltip: Tooltip,
+  },
+  props: {
+    modelValue: {
+      type: Array as PropType<Array<string>>,
+      default: () => [],
+    },
+  },
+  components: { Tree, OverlayPanel, InputText, Checkbox },
+  setup() {
+    return {
+      overlayPanel: ref<OverlayPanel>(),
+    };
+  },
+  methods: {
+    /**
+     * Executed, whenever the search bar input is focused. Opens the Tree Overlay.
+     * @param event the onclick event
+     */
+    inputFocused(event: Event) {
+      this.overlayPanel.show(event);
+    },
+    /**
+     * Handles the click event of the selection checkboxes
+     * @param key The key of the node whose checkbox has been clicked
+     * @param isChecked the new value of the checkbox
+     */
+    handleNodeCheckboxClick(key: string, isChecked: boolean) {
+      if (isChecked) this.selectedTreeNodes.add(key);
+      else this.selectedTreeNodes.delete(key);
+    },
+    /**
+     * Recursively expands the given nodes and all its children up to a maximum of max expanded nodes
+     * @param node the node to expand
+     * @param dict the dict to store the keys of the expanded nodes to
+     */
+    expandNode(node: TreeNode, dict: { [id: string]: boolean }) {
+      // Cap expanded nodes to 100 to prevent performance issues that would arise when
+      // e.g. someone searches for "a" and everything expands
+      if (Object.keys(dict).length >= 100) return;
+      dict[node.key] = true;
+      for (const child of node.children) {
+        this.expandNode(child, dict);
+      }
+    },
+    /**
+     * Recalculates the selectedChildrenCounters for all elements based on the selectedNodeSet
+     * @param selectedTreeNodeSet the set of selected nodes
+     */
+    updateSelectedChildrenCounter(selectedTreeNodeSet: Set<string>) {
+      const newSelectedChildrenCounter = new Map<string, number>();
+      const childrenCounterPopulator = (node: TreeNode): void => {
+        const children = node.children || [];
+        let localSum = 0;
+        for (let i = 0; i < children.length; i++) {
```

Comment:
why not 
```
children.foreach(child => {
   ...
});
```

Reply:
changed to `for (const element of children) {` to remove the index numbers

Reply:
well - actually it's 
```
        children.forEach((child) => {
```
right now...